src/OrleansRuntime/MembershipService/MembershipOracle.cs

```diff
@@ -1132,9 +1134,9 @@ public bool CheckHealth(DateTime lastCheckTime)
 
         #endregion
 
-        private static IMembershipService GetOracleReference(SiloAddress silo)
+        private IMembershipService GetOracleReference(SiloAddress silo)
         {
-            return InsideRuntimeClient.Current.InternalGrainFactory.GetSystemTarget<IMembershipService>(Constants.MembershipOracleId, silo);
+            return this.grainFactory.GetSystemTarget<IMembershipService>(Constants.MembershipOracleId, silo);
```

Comment:
I've seen this in many places, and I believe at some point it might be good to have something like IGrainAccessor<T> that you can explicitly state in the constructor that this will require a specific system target or grain (explicit dependency principle). Anyway, not suggesting we do it right now.